### PR TITLE
Generator: Insert into certificates_projects table

### LINKED ISSUE #9317: `lxc config trust add --project does-not-exist --restricted` should not succeed
This should not succeed:

```
# lxc config trust add /tmp/client.crt --projects does-not-exist --restricted
Error: Not Found

# lxc config trust ls --format csv
client,client,sdeziel@simon-laptop,502cdfa1d3e2,"Jul 17, 2018 at 11:25am (UTC)","Jul 14, 2028 at 11:25am (UTC)"

# lxc config trust show 502cdfa1d3e2
name: client
type: client
restricted: true
projects: []
certificate: |
  -----BEGIN CERTIFICATE-----
  -----END CERTIFICATE-----
fingerprint: 502cdfa1d3e2a29b1a3e9657c7f587f71aaba77d8aeeb84274a7c0f3dc74dcb6
```

Additional information:
```
# snap list lxd
Name  Version      Rev    Tracking     Publisher   Notes
lxd   git-48866d0  21586  latest/edge  canonical✓  -
```

### PR DESCRIPTION
**Pull request by masnax**
Closes #9317 

Coincidentally, I actually noticed this while working on the generator! The `certificates_projects` table was always updated independently of the `certificates` table. The fix I've put here should take care of the problem for now, until I finalize my reworking of how the generator handles reference tables.

### CODE REVIEW

**Comment by stgraber** [comment]
@masnax 
```
lxd/db/certificates.mapper.go:348:9: c.UpdateCertificatesProjects undefined (type *ClusterTx has no field or method UpdateCertificatesProjects)
```

File: lxd/db/certificates.go [hunk 1]
```diff
@@ -103,6 +103,11 @@ func (cert *Certificate) ToAPI() api.Certificate {
 
 // UpdateCertificateProjects updates the list of projects on a certificate.
 func (c *ClusterTx) UpdateCertificateProjects(id int, projects []string) error {
+	// If there are no project names, there's nothing to do.
+	if len(projects) == 0 {
+		return nil
```
Comment (tomponline) [inline lxd/db/certificates.go:108]: Shouldn't this case either remove all entries for that certificate, or return an error?

**Review by tomponline — CHANGES_REQUESTED** [review]
See comment about `UpdateCertificateProjects`

**Comment by tomponline** [comment]
@masnax maybe outside of the scope for this change, but can `c.UpdateCertificateProjects` be altered/moved/reworked so it can be called inside the transaction of the certificate operation that is being done, so that if there is a fault with either the whole change is rolled back?

**Comment by masnax** [comment]
> @masnax maybe outside of the scope for this change, but can `c.UpdateCertificateProjects` be altered/moved/reworked so it can be called inside the transaction of the certificate operation that is being done, so that if there is a fault with either the whole change is rolled back?

My understanding was that this is already the case? `UpdateCertificateProjects` is a method on `ClusterTx` which represents one cluster transaction.

**Comment by tomponline** [comment]
> > @masnax maybe outside of the scope for this change, but can `c.UpdateCertificateProjects` be altered/moved/reworked so it can be called inside the transaction of the certificate operation that is being done, so that if there is a fault with either the whole change is rolled back?
> 
> My understanding was that this is already the case? `UpdateCertificateProjects` is a method on `ClusterTx` which represents one cluster transaction.

Indeed it is, I'd read it as `c *Cluster`

**Review by tomponline — APPROVED** [review]

**Pull request merged**